pvl/pvl_validate.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+"""A program for testing and validating PVL text.
+
+The ``pvl_validate`` program will read a file with PVL text (any of
+the kinds of files that :func:`pvl.load` reads) and will report
+on which of the various PVL dialects were able to load that PVL
+text, and then also reports on whether the ``pvl`` library can encode
+the Python Objects back out to PVL text.
+
+You can imagine some PVL text that could be loaded, but is not able
+to be written out in a particular strict PVL dialect (like PDS3
+labels).
+"""
+
+# Copyright 2020, ``pvl`` library authors.
+#
+# Reuse is permitted under the terms of the license.
+# The AUTHORS file and the LICENSE file are at the
+# top level of this library.
+
+import argparse
+import logging
+import sys
+
+import pvl
+from .lexer import LexerError
+from .grammar import PVLGrammar, ODLGrammar, ISISGrammar, OmniGrammar
+from .parser import ParseError, PVLParser, ODLParser, OmniParser
+from .decoder import PVLDecoder, ODLDecoder, OmniDecoder
+from .encoder import PVLEncoder, ODLEncoder, ISISEncoder, PDSLabelEncoder
+
+dialects = dict(PDS3=dict(parser=ODLParser(),
+                          grammar=ODLGrammar(),
+                          decoder=ODLDecoder(),
+                          encoder=PDSLabelEncoder()),
+                ODL=dict(parser=ODLParser(),
+                         grammar=ODLGrammar(),
+                         decoder=ODLDecoder(),
+                         encoder=ODLEncoder()),
+                PVL=dict(parser=PVLParser(),
+                         grammar=PVLGrammar(),
+                         decoder=PVLDecoder(),
+                         encoder=PVLEncoder()),
+                ISIS=dict(parser=OmniParser(),
+                          grammar=ISISGrammar(),
+                          decoder=OmniDecoder(),
+                          encoder=ISISEncoder()),
+                Omni=dict(parser=OmniParser(),
+                          grammar=OmniGrammar(),
+                          decoder=OmniDecoder(),
+                          encoder=PVLEncoder()))
+dialect_order = ['PDS3', 'ODL', 'PVL', 'ISIS', 'Omni']
+
+
+def arg_parser():
+    p = argparse.ArgumentParser(description=__doc__)
+    p.add_argument('-v', '--verbose', action='count', default=0,
+                   help='Will report the errors that are encountered.')
+    p.add_argument('file', nargs='+',
+                   help='file containing PVL text to validate.')
+    return p
+
+
+def main():
+    args = arg_parser().parse_args()
+
+    logging.basicConfig(format='%(levelname)s: %(message)s',
+                        level=(60 - 20 * args.verbose))
+
+    results_list = list()
+    for f in args.file:
+        pvl_text = pvl.get_text_from(f)
+
+        results = dict()
+
+        for k, v in dialects.items():
+            results[k] = pvl_flavor(pvl_text, k, v, f, args.verbose)
+
+        # results['PDS3'] = pvl_flavor(pvl_text,
```

Comment:
commented out blocks are not needed, it's a philosophical thing to me but I use the git commit history to preserve code fragments instead of just commenting out blocks

Reply:
Agreed, this is just me being sloppy.  Fixed in upcoming commit.